Cython/Utility/ObjectHandling.c

```diff
@@ -857,6 +857,9 @@ static CYTHON_INLINE PyObject* __Pyx_Py{{type}}_GetSlice(
             PyObject* src, Py_ssize_t start, Py_ssize_t stop) {
     Py_ssize_t length = Py{{type}}_GET_SIZE(src);
     __Pyx_crop_slice(&start, &stop, &length);
+    if (length == 0) {
+        return Py{{type}}_New(0);    
+    }
```

Comment:
I think the undefined behaviour can only apply to lists and not tuples (since tuple `ob_item` is an array rather than a pointer) so it's probably worth restricting the check?

```suggestion
{{if type=='List'}}
    if (length == 0) {
        return Py{{type}}_New(0);    
    }
{{endif}}
```